### PR TITLE
server: implement dynamic forward 

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Ref #8477.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]

**Review by okJiang — COMMENTED** [review]
can we add any tests for it?

**Comment by rleungx** [comment]
> can we add any tests for it?

This PR doesn't change too much logic when the config value is false. I will add some tests in the next few PRs if we support changing the configuration.

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -403,11 +403,30 @@ func (c *RaftCluster) checkSchedulingService() {
...
+				log.Info("TSO is provided by TSO server")
+				if !c.IsServiceIndependent(constant.TSOServiceName) {
+					c.SetServiceIndependent(constant.TSOServiceName)
+				}
+			}
+		}
 		return
 	}
 
 	if err := c.startTSOJobs(); err != nil {
```
Comment (okJiang) [inline server/cluster/cluster.go:429]: How about renaming it to `startTSOJobsIfNeed`? When `isAPIServiceMode=false`, we will call `startTSOJobs` every time. Starting jobs too much times is strange.

**Review by okJiang — COMMENTED** [review]
rest lgtm

**Review by okJiang — APPROVED** [review]

**Pull request merged**